source/shared/cpp/ObjectModel/MarkDownParser.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <iomanip>
+#include <list>
+#include "pch.h"
+#include "BaseCardElement.h"
+#include "Enums.h"
+
+namespace AdaptiveCards
+{
+class MarkDownParser
+{
+public:
+    MarkDownParser(const std::string &txt);
+
+    enum DelimiterType
+    {
+        Init,
+        Alphanumeric,
+        Puntuation,
+        Escape,
+        WhiteSpace,
+        Underscore = 0x8,
+        Asterisk   = 0x10,
+    };
+
+    enum EmphasisType
+    {
+        EmphasisItalic = 1,
+        EmphasisBold = 2,
+    };
+
+    enum EmphasisState
+    {
+        EmphasisDetected = 0x1,
+        EmphasisDetecting = 0x2,
```

Comment:
Can you add some comments here?  What is the difference between EmphasisDetected and EmphasisDetecting?

Reply:
Thanks!  Does this mean that if the state is EmphasisDetected that m_curPos is pointing at a either a * or a _?  And that EmphasisDetecting is in-between the start and end of an emphasis run?  Could we call them something EmphasisStart, Emphasis (InsideEmphasis?), EmphasisEnd, OutsideEmphasis?  The detected and detecting language is confusing to me, let me know if i'm missing a subtlety here.

---
In reply to: [154409330](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154409330) [](ancestors = 154409330)

Reply:
you are absolutely correct. i'll make change; thank you!

---
In reply to: [154430904](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154430904) [](ancestors = 154430904,154409330)